include/counter.h

```diff
@@ -132,19 +254,125 @@ static inline int counter_set_alarm(struct device *dev,
  *
  * @param dev Pointer to the device structure for the driver instance.
  *
- * @retval 1 if the counter interrupt is pending.
+ * @retval 1 if the counter any interrupt is pending.
  * @retval 0 if no counter interrupt is pending.
  */
 __syscall int counter_get_pending_int(struct device *dev);
 
 static inline int _impl_counter_get_pending_int(struct device *dev)
 {
-	struct counter_driver_api *api;
+	const struct counter_driver_api *api = dev->driver_api;
 
-	api = (struct counter_driver_api *)dev->driver_api;
 	return api->get_pending_int(dev);
 }
 
+/**
+ * @brief Function to convert microseconds to ticks.
+ *
+ * @param[in]  dev    Pointer to the device structure for the driver instance.
+ * @param[in]  us     Microseconds.
+ *
+ * @return Converted ticks. Ticks will be saturated if exceed 32 bits.
+ */
+static inline u32_t counter_us_to_ticks(struct device *dev, u64_t us)
```

Comment:
`dev` can point to `const struct` in all these functions that only use `config_info` from it.

Reply:
done